Sources/Ink/Internal/URLDeclaration.swift

```diff
@@ -5,12 +5,12 @@
 */
 
 internal struct URLDeclaration: Readable {
-    var name: Substring
+    var name: String
     var url: URL
 
     static func read(using reader: inout Reader) throws -> Self {
         try reader.read("[")
-        let name = try reader.read(until: "]")
+        let name = try reader.read(until: "]").lowercased()
```

Comment:
I think we should do this at the end (on line `18`), to avoid unnecessarily string copying if any of the subsequent parsing steps fail. So on line I'd just do `name.lowerCased()` when initializing the `URLDeclaration`. Probably a minor thing, but every ounce of performance matters 😄